code/modules/mob/living/carbon/carbon.dm

```diff
@@ -768,6 +768,8 @@
 
 		for(var/datum/disease/D in viruses)
 			D.cure(0)
+			
+	regenerate_limbs()

```

Comment:
yeah you can't use admin_revive for things other than admin reviving, it removes cuffs, regrow limbs and plenty of other stuff.